submission/programs/adae.R

```diff
@@ -0,0 +1,283 @@
+# Name: ADAE
+#
+# Label: Adverse Event Analysis Dataset
+#
+# Input: ae, adsl
+# Output: adae.xpt
+#
+# Developed by : [Author/Developer name]
+
+library(admiral)
+library(dplyr)
+library(tidyr)
+library(lubridate)
+library(stringr)
+library(diffdf)
+library(xportr)
+library(metacore)
+library(metatools)
+
+# ----------------------------------------------------------------------------#
+# Use e.g. haven::read_sas to read in .sas7bdat, or other suitable functions  #
+# as needed and assign to the variables below.                                #
+# For illustration purposes read in admiral test data                         #
+# ----------------------------------------------------------------------------#
+
+# ---------- #
+# read in AE #
+# ---------- #
+ae <- haven::read_xpt(file.path("sdtm", "ae.xpt"))
+suppae <- haven::read_xpt(file.path("sdtm", "suppae.xpt"))
+
+# ------------ #
+# read in ADSL #
+# ------------ #
+adsl <- haven::read_xpt(file.path("submission", "datasets", "adsl.xpt"))
+
+#----------------------------------------------------------------------------------------#
+# When SAS datasets are imported into R using haven::read_sas(), missing                 #
+# character values from SAS appear as "" characters in R, instead of appearing           #
+# as NA values. Further details can be obtained via the following link:                  #
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values #
+#----------------------------------------------------------------------------------------#
+ae <- convert_blanks_to_na(ae)
+adsl <- convert_blanks_to_na(adsl)
+
+#---------------------- #
+# ADAE derivation start #
+#---------------------- #
+
+#------------------------------------#
+# Read in specifications from define #
+#------------------------------------#
+## placeholder for origin=predecessor, use metatool::build_from_derived()
+metacore <- spec_to_metacore("adam/TDF_ADaM - Pilot 3 Team updated.xlsx", where_sep_sheet = FALSE, quiet = T)
```

Comment:
On my console:
Error in `fn()`:                                                                                                           
! Can't combine `..1$dataset` <character> and `..2$dataset` <logical>.
Run `rlang::last_error()` to see where the error occurred.

Reply:
Hmmm... Can you try to pull `2_dev_adae_laxamanaj` again?  When I run I get successful import.

```
> metacore <- spec_to_metacore("adam/TDF_ADaM - Pilot 3 Team updated.xlsx", where_sep_sheet = FALSE, quiet = T)
                                                                                                                          
 Metadata successfully imported
Loading in metacore object with suppressed warnings
```

Reply:
@SHAESEN2 or @thomas-neitmann If ok, please `Resolve conversation`.

Reply:
Still does not work. The problem is that valuelevel is empty and therefore the list can't be reduced by purrr

 doc$ValueLevel
# A tibble: 0 × 18
# … with 18 variables: Order <lgl>, Dataset <lgl>, Variable <lgl>, Where Clause <lgl>, Label <lgl>, Data Type <lgl>,
#   Length <lgl>, Significant Digits <lgl>, Format <lgl>, Mandatory <lgl>, Assigned Value <lgl>, Codelist <lgl>,
#   Origin <lgl>, Pages <lgl>, Method <lgl>, Predecessor <lgl>, Comment <lgl>, Developer Notes <lgl>

Reply:
You might not have the latest version of {metacore} installed @SHAESEN2. Works flawless for me.